### PR TITLE
Eigen version

### PR DESCRIPTION
**Pull request by fghoussen**
## Pull request purpose

Switch eigen version to latest one : https://github.com/opencollab/arpack-ng/pull/259#issuecomment-636797682. On top of #260 .

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,6 +40,14 @@ stages:
 
 jobs:
   include:
+  # fedora (released fedora with openmpi)
```
Comment (sylvestre) [inline .travis.yml:43]: fedora / bionic 
please add a comment explaining why we mix the two :)

Comment (fghoussen) [reply .travis.yml:43]: no idea why ! not me !

**Comment by fghoussen** [comment]
CI can' take it.
> Get:37 http://archive.ubuntu.com/ubuntu xenial/universe amd64 libeigen3-dev all 3.3~beta1-2 [663 kB]
> ...
> checking for EIGEN3... no
> configure: error: Eigen not found. Give hints with PKG_CONFIG_PATH. Disable the need of this dependency using --disable-icb-exmm

@turboencabulator : I guess this is because CI provides "only 3.3.beta ? Do you kow how to overpass that ? If no, I'll close this PR.

File: .travis.yml (hunk 1 above)
Comment (dschwoerer) [reply .travis.yml:43]: Not sure what the question is?
Why fedora? Why openmpi? Why released?

I added an openmpi and an mpich case, to test both mpi versions, and also an released and still-in-development version of fedora, also to increase coverage. Why I mixed the on with the other, no real reason.

Comment (fghoussen) [reply .travis.yml:43]: Not sure what the question was either ! Good for me.

**Comment by fghoussen** [comment]
No idea with this fails. Seems unrelated with eigen version...

**Comment by fghoussen** [comment]
WTF ?!...

> Scanning dependencies of target arpackmm
830[ 64%] Building CXX object CMakeFiles/arpackmm.dir/EXAMPLES/MATRIX_MARKET/arpackmm.cpp.o
831virtual memory exhausted: Cannot allocate memory

**Comment by fghoussen** [comment]
Eigen 3.3. is available with bionic but trigger "831virtual memory exhausted: Cannot allocate memory" error. Eigen 3.3 is not available with xenial : cmake/autotools break, but, CI is OK.

**Comment by fghoussen** [comment]
@sylvestre : only solution I see for now, suppress `exmm` option for coverage in CI. May lower CI a (just) bit (`exmm` => Eigen Template => need lot of memory to compile + longer compile time) : your decision

**Comment by fghoussen** [comment]
@sylvestre : pass if you accept a small coverage decrease. Your decision !

**Comment by dschwoerer** [comment]
Another option to decrease memory consumption is decrease debugging level. You could try to use `-g1` rather then `-g`.

**Comment by sylvestre** [comment]
please update the changelog when you have a chance :)

**Pull request merged**